docs/internet-identity-spec.adoc

```diff
@@ -179,53 +179,9 @@ This section describes the interface that the backend canister provides.
 
 This interface is currently only used by its own frontend. This tight coupling means that this interface may change, even in incompatible ways. We therefore do not have to apply Candid best practices for backward-compatibility (such as using records for arguments and results).
 
-The summary is given by the following Candid interface (exluding the methods required for the https://www.notion.so/Design-HTTP-Requests-to-Canisters-d6bc980830a947a88bf9148a25169613[HTTP Gateway interface]):
+The summary is given by the following Candid interface:
 ....
-type UserNumber = nat64;
-type PublicKey = blob;
-type CredentialId = blob;
-type DeviceAlias = text;
-type DeviceKey = PublicKey;
-type UserKey = PublicKey;
-type SessionKey = PublicKey;
-type FrontendHostname = text;
-type Timestamp = nat64;
-
-type DeviceData = record {
-  pubkey : DeviceKey;
-  alias : text;
-  credential_id : opt CredentialId;
-  purpose : variant { recovery; authentication };
-  key_type : variant { unknown; platform; cross_platform; seed_phrase };
-};
-
-type Delegation = record {
-  pubkey: SessionKey;
-  expiration: Timestamp;
-  targets: opt vec principal;
-};
-
-type SignedDelegation = record {
-  delegation: Delegation;
-  signature: blob;
-};
-
-type GetDelegationResponse = variant {
-  signed_delegation: SignedDelegation;
-  no_such_delegation;
-};
-
-service : {
-  init_salt : () -> ();
-  register : (DeviceData) -> (UserNumber);
-  add : (UserNumber, DeviceData) -> ();
-  remove : (UserNumber, DeviceKey) -> ();
-  lookup : (UserNumber) -> (vec DeviceData) query;
-  get_principal : (UserNumber, FrontendHostname) -> (principal) query;
-
-  prepare_delegation : (UserNumber, FrontendHostname, SessionKey, maxTimeToLive : opt nat64) -> (UserKey, Timestamp);
-  get_delegation: (UserNumber, FrontendHostname, SessionKey, Timestamp) -> (GetDelegationResponse) query;
-}
+include::../src/internet_identity/internet_identity.did[]
```

Comment:
nice

Reply:
Hmm, it prevents being able to update the spec independently of the code (e.g. design a feature in one PR and implement it later) or to keep legacy or convenience stuff (e.g. monitoring) in the implementation that's not in the spec. At least that was the reasoning back then, maybe less of a problem now.